src/index.ts

```diff
@@ -99,20 +102,30 @@ const start = async () => {
     "interactionCreate",
     async (interaction: Interaction<CacheType>) => {
       //logger.debug({interaction});
-      if (!interaction.isCommand()) return;
-
-      const command = client.commands.get(interaction.commandName);
-
-      if (!command) return;
-
-      try {
-        await command.execute(interaction);
-      } catch (error) {
-        logger.error(error);
-        return interaction.reply({
-          content: "There was an error while executing this command!",
-          ephemeral: true,
-        });
+      if (interaction.isCommand()) {
```

Comment:
A better pattern here would be to extract the interaction type specific actions into their own functions.